snuba/migrations/table_engines.py

```diff
@@ -0,0 +1,59 @@
+from typing import Mapping, Optional
+
+
+class TableEngine:
+    pass
+
+
+class MergeTree(TableEngine):
```

Comment:
I think you could make these classes (you are going to build a few of them likely) dataclasses, so they are more compact and you do not need to provide a constructor each time.

Reply:
I'd rather not do this since most of the MergeTree parameters are currently optional but parameters of the children classes are not. All the other non-mergetree family table engines won't be dataclasses.

Reply:
ok